MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/MagicWindowActivity.kt

```diff
@@ -0,0 +1,305 @@
+package com.mapbox.mapboxandroiddemo.examples.labs
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.pm.PackageManager
+import android.graphics.*
+import android.location.Location
+import android.os.Build
+import android.os.Bundle
+import android.support.annotation.RequiresApi
+import android.support.v4.app.ActivityCompat
+import android.support.v4.content.ContextCompat
+import android.support.v7.app.AppCompatActivity
+import android.util.AttributeSet
+import android.util.TypedValue
+import android.view.MotionEvent
+import android.view.View
+import android.widget.FrameLayout
+import com.mapbox.android.core.location.LocationEngine
+import com.mapbox.android.core.location.LocationEngineListener
+import com.mapbox.android.core.location.LocationEnginePriority
+import com.mapbox.android.core.location.LocationEngineProvider
+import com.mapbox.mapboxandroiddemo.R
+import com.mapbox.mapboxsdk.Mapbox
+import com.mapbox.mapboxsdk.camera.CameraPosition
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import kotlinx.android.synthetic.main.activity_magic_window.*
+import kotlin.math.max
+import kotlin.math.min
+
+const val TAG = "DragActivityTag"
+const val PERMISSION_REQUEST_LOCATION = 404
+const val BASE_MAP_BUNDLE = "$TAG.basemap.bundle"
+const val REVEAL_MAP_BUNDLE = "$TAG.revealedMap.bundle"
```

Comment:
For these `const val`, please move them within the activity below the `var initialZoom` line with 

```kotlin
companion object {
        const val TAG = "DragActivityTag"
        const val PERMISSION_REQUEST_LOCATION = 404
        const val BASE_MAP_BUNDLE = "$TAG.basemap.bundle"
        const val REVEAL_MAP_BUNDLE = "$TAG.revealedMap.bundle"
}
```